### ISSUE TITLE
Changing a variable's name causes its type to desynchronize with the type in the type view

### ISSUE THREAD

**Issue by jeffli678** [Vector35/binaryninja-api#2428]
reported by yrp

**To Reproduce**

A very simple test script:

```C
#include <stdio.h>
#include <inttypes.h>
#include <stdlib.h>

struct A
{
    uint64_t a;
    uint64_t b;
};

int main()
{
    struct A* var = malloc(sizeof(struct A));
    var->a = 0x1234;
    var->b = 0x5678;
    return 0;
}
```

Create the type struct A and apply it to the variable in main:

![Screenshot from 2021-05-04 13-31-06](https://user-images.githubusercontent.com/3380590/116964023-02396d80-acdd-11eb-8f3d-8a14f1b5a47b.png)

Now, first rename the variable, e.g., to `foo`. And then go to type view, and edit the type struct A, e.g., by undefining a member. Then go back to the graph view, and we will see the undefined field still has its name showing up. And any further edits to the type in the type view is not reflected in the graph view. 

The reason, I believe, is the type of the variable is no longer the same one as the one in the type view, despite they have the same name. I need to test it and confirm this, though

Note, this has nothing to do with missing analysis update. It is persistent, that is not fixed by reanalyzing or anything

[desync.bndb.txt](https://github.com/Vector35/binaryninja-api/files/6419124/desync.bndb.txt)

**Comment by jeffli678** [comment]
Some extra information, after I undefine field `b` in the type view:

```Python
>>> bv.get_type_by_name('A').structure.members
[<uint64_t a, offset 0x0>]

>>> current_function.vars[1].type.element_type.structure.members
[<uint64_t a, offset 0x0>, <uint64_t b, offset 0x8>]

>>> bv.get_type_by_name('A') == current_function.vars[1].type.element_type
False
```

**Comment by CouleeApps** [comment]
This specific repro case starts happening with the build that adds HLIL array support ~2.3.2760-dev

**Comment by CouleeApps** [comment]
Fixed in 2.3.2819-dev
